internal/discovery/discovery.go

```diff
@@ -300,15 +301,21 @@ func (d *Discovery) SetParseOptions(options []hclparse.Option) {
 // WithOptions ingests runner options and applies any discovery-relevant settings.
 // Currently, it extracts HCL parser options provided via common.ParseOptionsProvider
 // and forwards them to discovery's parser configuration.
-func (d *Discovery) WithOptions(opts ...common.Option) *Discovery {
+func (d *Discovery) WithOptions(opts ...interface{}) *Discovery {
 	var parserOptions []hclparse.Option
 
 	for _, opt := range opts {
-		if p, ok := opt.(common.ParseOptionsProvider); ok {
+		if p, ok := opt.(interface{ GetParseOptions() []hclparse.Option }); ok {
 			if po := p.GetParseOptions(); len(po) > 0 {
 				parserOptions = append(parserOptions, po...)
 			}
 		}
+
+		if g, ok := opt.(interface{ GraphTarget() string }); ok {
+			if target := g.GraphTarget(); target != "" {
+				d = d.WithGraphTarget(target)
```

Comment:
does this mean if multiple opts with a `GraphTarget()` are passed in the last one wins and the earlier ones are thrown out? Is that the behavior we want?

Reply:
just double checking on this question